packages/code/src/renderingPlugin.tsx

```diff
@@ -151,46 +175,131 @@ export function LineTableRow({
   );
 }
 
+interface FlatTokenObject {
+  kind: string;
+  children: Array<string>;
+}
+
+// Check if object is a TokenObject which has an array with a single string element within it.
+function isFlattenedTokenObject(obj: TokenObject): obj is FlatTokenObject {
+  // default to an empty object in the off-chance "obj" is null or undefined.
+  const { children } = obj ?? {};
+
+  if (isArray(children) && children.length === 1 && isString(children[0])) {
+    return true;
+  }
+
+  return false;
+}
+
+// If an array of tokens contains an object with more than one children, this function will flatten that tree recursively.
+function flattenNestedTree(
+  children: Array<string | TokenObject>,
+  kind?: string,
+): Array<string | FlatTokenObject> {
+  if (typeof children === 'string') {
+    return children;
+  }
+
+  return children.reduce((acc, val) => {
+    if (isString(val)) {
+      // If there's a kind, we construct a custom token object with that kind to preserve highlighting.
+      // Without this, the value will simply render without highlighting.
+      const child = kind
+        ? { kind: generateKindClassName(kind), children: [val] }
+        : val;
+
+      return [...acc, child];
+    }
+
+    if (val?.children?.length > 1) {
```

Comment:
I think it's kinda sketchy to potentially compare `undefined > 1`, even if the behavior is well-defined in JS. how about doing `(val?.children?.length ?? 0) > 1` to make it more explicit

Reply:
Seems a bit unnecessary, but I don't feel strongly. I'll update it.